packages/core/src/lib/pseudopattern_hunter.js

```diff
@@ -60,10 +60,18 @@ pseudopattern_hunter.prototype.find_pseudopatterns = function(
       }
 
       //extend any existing data with variant data
-      variantFileData = _.merge(
+      variantFileData = _.mergeWith(
         {},
         currentPattern.jsonFileData,
-        variantFileData
+        variantFileData,
+        (objValue, srcValue) => {
+          if (
+            _.isArray(objValue) &&
+            patternlab.config.patternMergeVariantArrays
+          ) {
+            return objValue.concat(srcValue);
```

Comment:
If I'm not completely misstaken the PL2-way of doing this wasn't to concat but to replace at the key, so i guess this should be "return srcValue;" to only take the variant data and drop the parent array.

Reply:
Now I'm confused by myself. I need to test again in the evening to give an answer about what is happening here. I also should add a comment on that lines for other devs in the future.

Reply:
```
const mergeData = function (obj1, obj2) {
  /*eslint-disable no-param-reassign, guard-for-in*/
  if (typeof obj2 === 'undefined') {
    obj2 = {};
  }
  for (var p in obj1) {
    try {
      // Only recurse if obj1[p] is an object.
      if (obj1[p].constructor === Object) {
        // Requires 2 objects as params; create obj2[p] if undefined.
        if (typeof obj2[p] === 'undefined') {
          obj2[p] = {};
        }
        obj2[p] = mergeData(obj1[p], obj2[p]);

        // Pop when recursion meets a non-object. If obj1[p] is a non-object,
        // only copy to undefined obj2[p]. This way, obj2 maintains priority.
      } else if (typeof obj2[p] === 'undefined') {
        obj2[p] = obj1[p];
      }
    } catch (e) {
      // Property in destination object not set; create it and set its value.
      if (typeof obj2[p] === 'undefined') {
        obj2[p] = obj1[p];
      }
    }
  }
  return obj2;
};
```

This is the old merge method. It just recurses the parent object and sets every nested key if not already present in variant. So variant data always has priority and will never be overriden/merged/concatenated by the parent.

Example:

let parent = {a : 1, b : [2 , 3], c : {d : [4, 5], e : 8}};
let variant = {a : 2, c : {d : [6, 7]}};
mergeData(parent, variant);
=> {a : 2, b : [2 , 3], c : {d : [6, 7], e : 8}};

Reply:
That is what I'm also known to from 2.x. But currently, it is exactly the opposite of the current version. 

So the default should be that variants arrays merge and that you can turn it off.

Now that I tested the mechanism again, it is correct that the functionality was not correct. It should be like the following code. (I pushed an update)

```javascript
if (
  _.isArray(objValue) &&
  !patternlab.config.patternMergeVariantArrays
) {
  return srcValue;
}
```